File: safehere_flutterapp/lib/models/trusted_contact.dart
class TrustedContact {
  final String id;
  final String name;
  final String relationship;
  final String? phone;
  final String? email;
  final String avatarColor;
  final String status; // pending | verified | active
  final bool online;
  final bool locationSharing;
  final bool isEmergencyContact;

  TrustedContact({
    required this.id,
    required this.name,
    required this.relationship,
    this.phone,
    this.email,
    required this.avatarColor,
    required this.status,
    required this.online,
    required this.locationSharing,
    required this.isEmergencyContact,
  });

  factory TrustedContact.fromJson(Map<String, dynamic> json) => TrustedContact(
        id: json['id'],
        name: json['name'],
        relationship: json['relationship'] ?? '',
        phone: json['phone'],
        email: json['email'],
        avatarColor: json['avatar_color'] ?? '#9B8AFB',
        status: json['status'] ?? 'pending',
        online: json['online'] ?? false,
        locationSharing: json['location_sharing'] ?? false,
        isEmergencyContact: json['is_emergency_contact'] ?? true,
      );
}

File: backend_safeher/app/models/trusted_contact.py
import uuid
from datetime import datetime, timezone
import enum

from beanie import Document
from pydantic import Field
from pymongo import IndexModel


def _uuid() -> str:
    return str(uuid.uuid4())


class ContactStatus(str, enum.Enum):
    pending = "pending"
    verified = "verified"
    active = "active"


class TrustedContact(Document):
    id: str = Field(default_factory=_uuid)
    owner_id: str

    name: str
    relationship: str = ""
    phone: str | None = None
    email: str | None = None
    avatar_color: str = "#9B8AFB"

    status: ContactStatus = ContactStatus.pending
    online: bool = False
    location_sharing: bool = False
    is_emergency_contact: bool = True

    # set once the invited email matches a registered SafeHer user
    linked_user_id: str | None = None

    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "trusted_contacts"
        indexes = [
            IndexModel("owner_id"),
        ]
